junifer/storage/hdf5.py

```diff
@@ -819,116 +878,113 @@ def collect(self) -> None:
-                # Write chunked array
-                data_to_write["data"] = ChunkedArray(
-                    data=features_data,
-                    shape=tuple(array_shape),
-                    chunk_size=tuple(array_chunk_size),
-                    n_chunk=chunk_idx,
-                )
-                # Write to HDF5
-                write_hdf5(
-                    fname=str(self.uri.resolve()),  # type: ignore
-                    data=data_to_write,
-                    overwrite=self.overwrite,  # type: ignore
-                    compression=0,
+
+                # Read the data
+                t_data = read_hdf5(
+                    fname=str(file_),
                     title=feature_md5,
-                    slash="error",
-                    use_json=False,
+                    slash="ignore",
                 )
+                if i_file == 0:
+                    # Store the "static" data
+                    static_data = {
+                        k: v
+                        for k, v in t_data.items()
+                        if k not in ["data", "element"]
+                    }
+                    kind = static_data["kind"]
+
+                # Append the "dynamic" data
+                if kind == "timeseries":
+                    chunk_data.extend(t_data["data"])
+                else:
+                    chunk_data.append(t_data["data"])
+                elements.extend(t_data["element"])
+
+                i_file += 1
+                if (i_file % t_chunk_size == 0) or i_file == element_count:
+                    # If we have reached the chunk size or the end of the
+                    # elements, write the data
+
+                    # Store one chunk of data
+                    to_write = static_data.copy()
+                    to_write["element"] = []
+                    # Write data in chunks to avoid memory usage spikes
+                    # Start with the case for 2D
+                    # Write chunked array
+                    to_write["data"] = _create_chunk(
+                        chunk_data=chunk_data,
+                        kind=kind,
+                        element_count=element_count,
+                        chunk_size=t_chunk_size,
+                        i_chunk=i_chunk,
+                    )
+                    if i_file == element_count:
+                        to_write["element"] = elements
+
+                    # Write to HDF5
+                    write_hdf5(
+                        fname=str(self.uri.resolve()),  # type: ignore
+                        data=to_write,
+                        overwrite="update",  # type: ignore
+                        compression=0,
+                        title=feature_md5,
+                        slash="error",
+                        use_json=False,
+                    )
+
+                    # Increment counters and data
+                    i_chunk += 1
+                    chunk_data = []
```

Comment:
Would be a safe measure to `del to_write` to further reduce memory usage.

Reply:
Let the GC do it when it considers it necessary. For small datasets, this might increase the computation time.